### ISSUE TITLE
Breaking change with requiring `npm 7` without bumping the major version

### ISSUE THREAD

**Issue by gpinto67** [apollographql/apollo-client#8577]
This PR https://github.com/apollographql/apollo-client/commit/68d9cabab976a4ca659e161ec52ea1700aab8496 is causing issues with our repo. 
```
➜  unified-checkout-v3 git:(gpinto/ECOMM-106839_sessionTimeOut_emitMessage) ✗ npm i
npm ERR! code ENOTSUP
npm ERR! notsup Unsupported engine for @apollo/client@3.4.3: wanted: {"npm":"^7.20.3"} (current: {"node":"12.14.1","npm":"6.14.13"})
npm ERR! notsup Not compatible with your version of node/npm: @apollo/client@3.4.3
npm ERR! notsup Not compatible with your version of node/npm: @apollo/client@3.4.3
npm ERR! notsup Required: {"npm":"^7.20.3"}
npm ERR! notsup Actual:   {"npm":"6.14.13","node":"12.14.1"}
```
Why enforcing the npm version?

**Comment by benjamn** [comment]
@gpinto67 As the commit message mentions, 68d9cabab976a4ca659e161ec52ea1700aab8496 was only intended to help @renovate figure out which version of `npm` to use when generating dependency update PRs. If I'm understanding correctly, this means we could keep the `engines.npm` section in the root `package.json` file (since that's what @renovate reads), and then remove it when we generate `dist/package.json` (which is what gets published to npm). That should prevent `engines.npm` from imposing any constraints on code using `@apollo/client` . Apollo Client currently has no reason to require you to use npm v7.

**Comment by gpinto67** [comment]
In theory that should work, and unbreak for us. For now I locked my dependency to `@apollo/client": "3.3.11"` (no semver `^` marker).

**Comment by benjamn** [comment]
Fix incoming: #8578

**Comment by gpinto67** [comment]
🥇  Thanks for the quick turnaround!
